test/test_model.py

```diff
@@ -50,20 +56,79 @@ def setUp(self):
 
     def test_model(self):
         self.assertIs(self.model.app, self.model.unit.app)
+
+    def test_relations_keys(self):
         for relation in self.model.relations['db2']:
             self.assertIn(self.model.unit, relation.data)
             unit_from_rel = next(filter(lambda u: u.name == 'myapp/0', relation.data.keys()))
             self.assertIs(self.model.unit, unit_from_rel)
+
+    def test_relation_helper(self):
         self.assertIsNone(self.model.relation('db0'))
         self.assertIsInstance(self.model.relation('db1'), juju.model.Relation)
         with self.assertRaises(juju.model.TooManyRelatedApps):
             self.model.relation('db2')
+
+    def test_relation_data(self):
         random_unit = self.model._cache.get(juju.model.Unit, 'randomunit/0')
         with self.assertRaises(KeyError):
             self.model.relation('db1').data[random_unit]
         remoteapp1_0 = next(filter(lambda u: u.name == 'remoteapp1/0', self.model.relation('db1').units))
         self.assertEqual(self.model.relation('db1').data[remoteapp1_0], {'host': 'remoteapp1-0'})
 
+    def test_relation_data_modify_remote(self):
+        rel_db1 = self.model.relation('db1')
+        backend = self.model._backend
```

Comment:
This is our test backend. We shouldn't have to poke into private details of the implementation to get back to it.